src/thunderbots/software/ai/hl/stp/evaluation/indirect_chip.h

```diff
@@ -0,0 +1,121 @@
+#include "ai/world/world.h"
+#include "geom/angle.h"
+#include "geom/point.h"
+#include "util/parameter/dynamic_parameters.h"
+
+/**
+ * The indirect chip and chase target evaluation function returns a target point for the
+ * chipper and chaser.
+ * Other related evaluation functions are also included in this file.
+ */
+namespace Evaluation
+{
+    typedef Point Vector2;
+    template <size_t N>
+    using Poly     = std::array<Vector2, N>;
+    using Triangle = Poly<3>;
+
+    /**
+     * Returns the target point that the chipper will shoot at and the chaser will meet
+     * ball at. Given the enemy team robot's positions, filters and creates a vector of
+     * triangles to be used as a parameter. The target is where ball will land according
+     * to chipping calibration.
+     *
+     * @param World Object
+     *
+     * @return Point to chip and chase at; If empty, point is not valid
```

Comment:
Added clarifications on "not valid": 
Since the `valid` boolean is now replaced by the `optional`, I decided to call "not valid" to be "null" instead. 

The point would be "null" if the target triangles were empty.  However, I'm actually a little confused too about what "empty" means. The target triangles are supposed to be filtered so that they are open triangles that do not consist of enemy robots. Wouldn't that mean the triangles are now empty?

Reply:
- if we wanted to be super clear, we could say `std::nullopt` (which is usually what we've done elsewhere), but I'm fine with just `null` as well
- Hmm, I think "empty" here is almost certainly referring to the list of triangles being empty (ie. we could not find any triangles. 
- The caller of this function probably doesn't care about triangles though, they just want a point to chip to. In general, this touches on the rule that _interface should be separate from implementation_. Basically, as someone else who uses this function, I shouldn't have to know anything about how it works. Keeping this in mind, I'd suggest updating this comment to something like `Point to chip and chase at; If std::nullopt, we could find no good point to chip and chase to`

Reply:
Edited as suggested! That makes sense. Say if I didn't know what this function was for, I'd have no idea what "empty triangles" referred to.

Reply:
yep!